File: src/services/app.service.ts
import { callTauri } from "./tauri";

export const appService = {
  /**
   * 获取本地 ~/.monster-tools 应用专属数据根目录
   */
  async getAppPaths(): Promise<string> {
    return await callTauri<string>("get_app_paths");
  },

  /**
   * 获取底座包版本号
   */
  async getAppVersion(): Promise<string> {
    return await callTauri<string>("get_app_version");
  }
};

File: src-tauri/src/services/app_service.rs
use tauri::AppHandle;
use crate::infra::path::PathProvider;
use crate::infra::AppResult;

pub struct AppService {
    app_handle: AppHandle,
    path_provider: PathProvider,
}

impl AppService {
    pub fn new(app_handle: AppHandle, path_provider: PathProvider) -> Self {
        Self { app_handle, path_provider }
    }

    /// 获取当前应用版本号
    pub fn get_version(&self) -> String {
        self.app_handle.package_info().version.to_string()
    }

    /// 获取本地 ~/.monster-tools 数据存放物理路径
    pub fn get_local_data_dir(&self) -> AppResult<String> {
        let path = self.path_provider.get_app_local_data_dir()?;
        Ok(path.to_string_lossy().into_owned())
    }
}
